### DISCUSSION TITLE
Taking a dependency on cxplat only

### DISCUSSION THREAD

**Discussion by nigriMSFT** [microsoft/usersim#167] [Q&A] [answered]
I am working in a [repository](https://github.com/microsoft/xdp-for-windows) that now has a need for a cxplat abstraction very similar to what is provided in this repository. My usage only needs support for Windows user mode and Windows kernel mode, as the project is scoped to Windows.

Is taking a dependency on this usersim repository only for the cxplat facilities it provides a good/reasonable idea?

Another route we can take is to copy cxplat facilities from usersim or [msquic](https://github.com/microsoft/msquic/tree/main/src/platform), but I'd really hope to not create yet another Windows cxplat library.

There is also [cxplat](https://github.com/microsoft/cxplat), are/were there plans to make this a centralized cxplat facility?

**Comment by nibanks** [comment, marked as answer]
The goal is that [cxplat](https://github.com/microsoft/cxplat) would be **the** place for cross-platform abstraction, and everything would depend on it. But since that requires quite a bit of work, and we haven't had a lot of justification for it, we haven't gotten there yet.

**Reply by nigriMSFT** [reply]
Thanks for the context!
